src/github.com/couchbase/sync_gateway/db/index_changes.go

```diff
@@ -168,13 +117,19 @@ func (db *Database) VectorMultiChangesFeed(chans base.Set, options ChangesOption
 				if minEntry.Seq.TriggeredBy == 0 {
 					// Update the cumulative clock, and stick it on the entry.
 					cumulativeClock.SetMaxSequence(minEntry.Seq.vbNo, minEntry.Seq.Seq)
-					// TODO: Potential enhancement - for first iteration loop, always set hash on last entry.
-					continuousLastHash = db.calculateHashWhenNeeded(
+					// Force new hash generation for non-continuous changes feeds if this is the last entry to be sent - either
+					// because there are no more entries in the channel feeds, or we're going to hit the limit.
+					forceHash := false
+					if options.Continuous == false && (nextEntry == nil || options.Limit == 1) {
+						forceHash = true
```

Comment:
Ok, after seeing this:

```
				// Stop when we hit the limit (if any):
				if options.Limit > 0 {
					options.Limit--
					if options.Limit == 0 {
						break outer
					}
				}
```

the `options.Limit == 1` check makes a bit more sense.  It's fairly misleading though, because when I see `options.Limit` I'm assuming that's an immutable parameter passed in by the user.  I think renaming it to `limitMarker` and initializing it to `options.Limit` at the top of the function would make the code clearer.